From b80c01d829a63c0998fac91bebbc613df95f21ea Mon Sep 17 00:00:00 2001
From: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Date: Thu, 13 May 2010 19:28:30 -0400
Subject: [PATCH 364/391] tsc-sync-relax-check

tsc sync relax check

At some point, when booting my 8-core Xeon, 1 out of 8 CPUs seems to be 1700
cycles apart from the others. It might be caused by more frequent SMIs, or by
the CPU just being a bit off.

So let's increase the number of test loops from 10 to 200 (to account for more
frequent SMIs) and relax the accepted range to from 1000 to 3000 cycles.

Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
---
 kernel/time/tsc-sync.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/kernel/time/tsc-sync.c b/kernel/time/tsc-sync.c
index fc181d8..2ac1544 100644
--- a/kernel/time/tsc-sync.c
+++ b/kernel/time/tsc-sync.c
@@ -34,12 +34,12 @@
 #include <linux/mutex.h>
 #include <linux/cpu.h>
 
-#define MAX_CYCLES_DELTA 1000ULL
+#define MAX_CYCLES_DELTA 3000ULL
 
 /*
  * Number of loops to take care of MCE, NMIs, SMIs.
  */
-#define NR_LOOPS	10
+#define NR_LOOPS	200
 
 static DEFINE_MUTEX(tscsync_mutex);
 
-- 
1.6.5.2

